src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -399,124 +397,13 @@ fsm_resume_as_primary(Keeper *keeper)
  * fsm_prepare_replication is used when a new standby was added.
  *
  * add_standby_to_hba && create_replication_slot
+ *
+ * Those operations are now done eagerly rather than just in time. So it's been
+ * taken care of aready, nothing to do within this state transition.
```

Comment:
It would be possible yes. At the moment having a function that only does `return true` or a NULL function pointer in the KeeperFSM structure bears the same meaning. I though that we might want to do more things in that transition later though, and keeping that around does not sound like a maintenance burden...